### DISCUSSION TITLE
Is there a way to get suggestions that are consistent with my set of Typescript types and GraphQL schema?

### DISCUSSION THREAD

**Discussion by benjaminLooper** [community/community#58577] [Copilot Conversations]
Is there a way to get suggestions that are consistent with my set of Typescript types and GraphQL schema? 

I'm working on a project that uses Typescript and GraphQL / Apollo. 
I have a frontend copy of the backend GraphQL schema, and I generate a large file of TS types based on that schema.
I've been using copilot and it's been awesome, but sometimes it makes suggestions that are inconsistent with my TS types (i.e. misnamed fields, type mismatches when assigning values to fields, etc). 

Furthermore, when I'm writing GraphQL queries / mutations it'd be nice if copilot was aware of the schema (the copy in the frontend). A use case for this could be to make a comment asking for all of the fields from a table that has 20+ fields (which is a pain), or to ask for a fragment that I've already defined elsewhere, or maybe even ask for fields of a certain type.

It could be that this already exists, or there's some configuration that I'm unaware of, but I wanted to put it out there.
Any thoughts or suggestions?

**Comment by baronyoung** [comment]
A similar question was asked in Feb with no answer so my assumption is this just isn't supported yet.  Hopefully it will be at some point.
